### PR TITLE
Fixes for some Linux crashes

### PR DESCRIPTION
**Pull request by xrombik**

### CODE REVIEW

File: src/Layers/xrRender/HW.h [hunk 1]
```diff
@@ -118,4 +118,4 @@ class CHW
 #endif // USE_OGL
 };
 
-extern ECORE_API CHW HW;
+//extern volatile ECORE_API CHW HW;
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/HW.h]: Just NO! This changes renderer structure. It's wrong, even if it changes it a bit of things. `extern HW` should be defined here.

File: src/Layers/xrRenderGL/glHW.cpp [hunk 2]
```diff
@@ -65,6 +66,8 @@ void CHW::CreateDevice(SDL_Window* hWnd)
         Msg("Could not initialize glew.");
         return;
     }
+    char *glew_ver = (char*) glewGetString(GLEW_VERSION);
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderGL/glHW.cpp]: Inconsistent code style.
Should be:
```cpp
char* glewVersion = (char*)glewGetString(GLEW_VERSION);
```

File: src/Layers/xrRenderGL/glHW.cpp [hunk 3]
```diff
@@ -74,6 +77,7 @@ void CHW::CreateDevice(SDL_Window* hWnd)
     Msg("* GPU vendor: [%s] device: [%s]", glGetString(GL_VENDOR), glGetString(GL_RENDERER));
     Msg("* GPU OpenGL version: %s", glGetString(GL_VERSION));
     Msg("* GPU OpenGL shading language version: %s", glGetString(GL_SHADING_LANGUAGE_VERSION));
+    Msg("* OpenGL Extension Wrangler: %s", glew_ver);
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderGL/glHW.cpp]: Can we do just like below?
```cpp
Msg("* OpenGL Extension Wrangler: %s", glewGetString(GLEW_VERSION));
```

File: src/Layers/xrRenderGL/glR_Backend_Runtime.h [hunk 4]
```diff
@@ -4,6 +4,8 @@
 
 #include "glStateUtils.h"
 
+extern CHW HW;
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderGL/glR_Backend_Runtime.h]: My comment about this is in HW.h

In addition, this brokes all DirectX renderers, since you didn't included this change in `R_Backend_Runtime.h` and `dx10R_Backend_Runtime.h`.

File: src/xrEngine/main.cpp [hunk 5]
```diff
@@ -198,7 +198,7 @@ ENGINE_API void Startup()
 
     // Main cycle
     Device.Run();
-    // Destroy APP
+   //  Destroy APP
```
Comment (Xottab-DUTY) [inline src/xrEngine/main.cpp]: Unrelated change with inconsistent code style

File: src/xrEngine/main.cpp [hunk 6]
```diff
@@ -209,12 +209,12 @@ ENGINE_API void Startup()
...
 #endif
-        destroySettings();
-    LALib.OnDestroy();
+     destroySettings();
+   LALib.OnDestroy();
 #if !defined(LINUX)
     if (!g_bBenchmark && !g_SASH.IsRunning())
 #endif
-        destroyConsole();
+       destroyConsole();
```
Comment (Xottab-DUTY) [inline src/xrEngine/main.cpp]: Unrelated change with inconsistent code style

File: src/xrEngine/main.cpp [hunk 7]
```diff
@@ -209,12 +209,12 @@ ENGINE_API void Startup()
 #if !defined(LINUX)
     if (!g_bBenchmark && !g_SASH.IsRunning())
 #endif
-        destroySettings();
-    LALib.OnDestroy();
+     destroySettings();
+   LALib.OnDestroy();
```
Comment (Xottab-DUTY) [inline src/xrEngine/main.cpp]: Unrelated change with inconsistent code style

File: src/xrEngine/xr_input.cpp [hunk 8]
```diff
@@ -195,7 +195,7 @@ pcstr KeyToMouseButtonName(const int dik)
 bool CInput::get_dik_name(int dik, LPSTR dest_str, int dest_sz)
 {
     xr_string keyname;
-    static std::locale locale("");
+    static std::locale locale("ru_RU.UTF-8");
```
Comment (Xottab-DUTY) [inline src/xrEngine/xr_input.cpp]: Just NO! The game is a world-wide product localized to several languages. We should not use hardcoded locales in the code.

File: src/xrEngine/xr_ioc_cmd.cpp [hunk 9]
```diff
@@ -264,7 +264,9 @@ void CCC_LoadCFG::Execute(LPCSTR args)
         {
             F->r_string(str, sizeof(str));
             if (allow(str))
+            {
                 Console->Execute(str);
+            }
```
Comment (Xottab-DUTY) [inline src/xrEngine/xr_ioc_cmd.cpp]: Unrelated change

File: src/xrEngine/xr_ioc_cmd.h [hunk 10]
```diff
@@ -223,6 +238,11 @@ class ENGINE_API CCC_Token : public IConsole_Command
             tips.push_back("--- (current)");
         }
         tok = GetToken();
+        if (!tok)
+        {
+            Msg("! token [%s] is null", cName);
+            return;
+        }
```
Comment (Xottab-DUTY) [inline src/xrEngine/xr_ioc_cmd.h]: Crash is supposed to happen. Replace this with:
```cpp
R_ASSERT3(tok, "Failed to get token", cName)
```

File: src/xrGame/string_table.cpp [hunk 11]
```diff
@@ -58,7 +61,6 @@ void CStringTable::Init()
 #ifdef DEBUG
     Msg("StringTable: loaded %d files", fset.size());
 #endif // #ifdef DEBUG
-
```
Comment (Xottab-DUTY) [inline src/xrGame/string_table.cpp:66]: Unrelated whitespace change

File: src/Layers/xrRenderGL/glHW.cpp [hunk 12]
```diff
@@ -10,6 +10,7 @@
 #include "Include/xrAPI/xrAPI.h"
 #include "xrCore/xr_token.h"
 
+
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderGL/glHW.cpp]: Unrelated whitespace change

File: src/xr_3da/entry_point.cpp [hunk 13]
```diff
@@ -42,7 +42,8 @@ int entry_point(pcstr commandLine)
     }
     Core.Initialize("OpenXRay", commandLine, nullptr, true, *fsgame ? fsgame : nullptr);
 
-    auto result = RunApplication();
+    int result = 0;
+    result = RunApplication();
```
Comment (Xottab-DUTY) [inline src/xr_3da/entry_point.cpp]: Unneeded change.

File: src/xr_3da/entry_point.cpp [hunk 14]
```diff
@@ -98,8 +100,7 @@ int main(int argc, char *argv[])
         }
 
         result = entry_point(commandLine);
-
-        free(commandLine);
+        if (commandLine) free(commandLine); {commandLine = NULL;};
```
Comment (Xottab-DUTY) [inline src/xr_3da/entry_point.cpp]: Inconsistent codestyle.
```suggestion
        if (commandLine)
            free(commandLine);
        commandLine = nullptr;
```
I wanted to say that it's prohibited to use standard functions and we should use `xr_free` instead of `free`. But as far as I know, xrMemory isn't initialized in this moment..

File: src/xr_3da/entry_point.cpp [hunk 15]
```diff
@@ -76,16 +77,17 @@ int APIENTRY WinMain(HINSTANCE inst, HINSTANCE prevInst, char* commandLine, int
...
     {
-        char* commandLine = "";
+
         int i;
         if(argc > 1)
         {
             size_t sum = 0;
             for(i = 1; i < argc; ++i)
-                sum += strlen(argv[i]) + 1;
+                sum += strlen(argv[i]) + strlen(" ") + sizeof('\0');
```
Comment (Xottab-DUTY) [inline src/xr_3da/entry_point.cpp]: `xr_strlen` instead of `strlen` should be used.

File: src/Layers/xrRenderGL/glSH_Texture.cpp [hunk 16]
```diff
@@ -208,13 +208,6 @@ void CTexture::Load()
 
             pSurface = pTexture;
             desc = GL_TEXTURE_2D;
-            GLenum err = glGetError();
-            if (err != GL_NO_ERROR)
-            {
-                FATAL_F("Invalid video stream: 0x%x", err);
-                xr_delete(pTheora);
-                pSurface = 0;
-            }
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderGL/glSH_Texture.cpp:217]: Maybe it's better to do like this?
```cpp
            GLenum err = glGetError();
            if (err != GL_NO_ERROR)
            {
                Msg("! Invalid video stream: 0x%x", err);
                // Sometimes it crashes if video stream is correct
                // XXX: Fix!
                //FATAL_F("Invalid video stream: 0x%x", err);
                //xr_delete(pTheora);
                //pSurface = 0;
            }
```

File: src/xrGame/string_table.cpp [hunk 17]
```diff
@@ -82,21 +85,24 @@ void CStringTable::FillLanguageToken()
...
+        if (languagesToken.size() > LanguageID)
+            pData->m_sLanguage = languagesToken.at(LanguageID).name;
     else
     {
         pData->m_sLanguage = pSettings->r_string("string_table", "language");
-        auto it = std::find_if(languagesToken.begin(), languagesToken.end(), [](const xr_token& token) {
-            return token.name && token.name == pData->m_sLanguage;
-        });
+        auto it = std::find_if(languagesToken.begin(), languagesToken.end(), [](const xr_token& token)
+            {return token.name && token.name == pData->m_sLanguage;});
```
Comment (Xottab-DUTY) [inline src/xrGame/string_table.cpp]: Unrelated change with inconsistent code style.
Should be this:
```suggestion
        {
            return token.name && token.name == pData->m_sLanguage;
        });
```

File: src/xr_3da/entry_point.cpp [hunk 18]
```diff
@@ -76,16 +77,17 @@ int APIENTRY WinMain(HINSTANCE inst, HINSTANCE prevInst, char* commandLine, int
 int main(int argc, char *argv[])
 {
     int result = EXIT_FAILURE;
-
+    static std::locale locale("");
```
Comment (Xottab-DUTY) [inline src/xr_3da/entry_point.cpp]: Wait.. Is this locale even used?

File: src/xrGame/string_table.cpp [hunk 19]
```diff
@@ -82,21 +85,24 @@ void CStringTable::FillLanguageToken()
 void CStringTable::SetLanguage()
 {
     if (LanguageID != std::numeric_limits<u32>::max())
-        pData->m_sLanguage = languagesToken.at(LanguageID).name;
+        if (languagesToken.size() > LanguageID)
+            pData->m_sLanguage = languagesToken.at(LanguageID).name;
```
Comment (Xottab-DUTY) [inline src/xrGame/string_table.cpp:105]: I will submit my refactoring changes, soon. I hope they won't collide with this..

**Review by Xottab-DUTY — CHANGES_REQUESTED** [review]
Thank you for the effort! Please, forgive me for such strict and hard review.

File: src/Layers/xrRenderGL/glSH_Texture.cpp (hunk 16 above)
Comment (xrombik) [reply src/Layers/xrRenderGL/glSH_Texture.cpp:217]: No. Error OpenGL appears earlier, and is not related in any way with the "video stream". Just a error or a few errors remained in the stack of errors, and one of them got here. Where an error (s) appeared, it is difficult to determine with the help of glGetError (), since it needs to be called after each gl * () call to know for sure. Thus is more broad Msg("! Some erroneous usage of OpenGL detected"); is good.

Comment (eagleivg) [reply src/Layers/xrRenderGL/glSH_Texture.cpp:217]: More right way is writing error handler with ability skip uncritical errors.

File: src/xrEngine/xr_input.cpp (hunk 8 above)
Comment (eagleivg) [reply src/xrEngine/xr_input.cpp]: Need just add getting locale from environment variable

Comment (xrombik) [reply src/xrEngine/xr_input.cpp]: Calling std::locale locale(""); - this is set default locale, while only very new libc set defaul locale without this hack.

File: Externals/ode/ode/src/timer.cpp [hunk 20]
```diff
@@ -374,45 +375,51 @@ void dTimerReport (FILE *fout, int average)
 
   // get maximum description length
   maxl = 0;
-  for (i=0; i<num; i++) {
+  for (i=0; i<num; ++i)
```
Comment (FreeZoneMods) [inline Externals/ode/ode/src/timer.cpp]: What about using standard algo for determining maxl?

File: src/Layers/xrRenderGL/glState.cpp [hunk 21]
```diff
@@ -234,7 +234,7 @@ void glState::UpdateSamplerState(u32 stage, u32 name, u32 value)
             value, currentFilter, true)));
         break;
     case D3DSAMP_MIPMAPLODBIAS: /* float Mipmap LOD bias */
-        CHK_GL(glSamplerParameterf(m_samplerArray[stage], GL_TEXTURE_LOD_BIAS, *(float*)value));
+        CHK_GL(glSamplerParameterf(m_samplerArray[stage], GL_TEXTURE_LOD_BIAS, (GLfloat)value));
```
Comment (FreeZoneMods) [inline src/Layers/xrRenderGL/glState.cpp]: Use static_cast, please

File: src/xrCore/Compression/SubAlloc.hpp [hunk 22]
```diff
@@ -287,7 +287,7 @@ inline void SpecialFreeUnit(void* ptr)
         BList->insert(ptr, 1);
     else
     {
-        *(DWORD*)ptr = ~0UL;
+        *(DWORD*)ptr = ~0U;
```
Comment (FreeZoneMods) [inline src/xrCore/Compression/SubAlloc.hpp:290]: Consider removing these ugly C-style casts, please

File: src/xrCore/Compression/SubAlloc.hpp [hunk 23]
```diff
@@ -63,7 +63,7 @@ inline void BLK_NODE::insert(void* pv, int NU)
 {
     MEM_BLK* p = (MEM_BLK*)pv;
```
Comment (FreeZoneMods) [inline src/xrCore/Compression/SubAlloc.hpp:64]: Use C++-style cast, please. And maybe add a kind of assertion here?

File: src/xrCore/Compression/SubAlloc.hpp [hunk 24]
```diff
@@ -63,7 +63,7 @@ inline void BLK_NODE::insert(void* pv, int NU)
...
     link(p);
-    p->Stamp = ~0UL;
+    p->Stamp = ~0U;
```
Comment (FreeZoneMods) [inline src/xrCore/Compression/SubAlloc.hpp:66]: Maybe use std::bitset instead of this bitmagic?

File: src/xrCore/FS.cpp [hunk 25]
```diff
@@ -181,7 +182,8 @@ void* FileDecompress(const char* fn, const char* sign, u32* size)
 
     int H = _open(fn, O_BINARY | O_RDONLY);
     R_ASSERT2(H > 0, fn);
-    _read(H, &F, 8);
+    int bytesRead = _read(H, &F, 8);
+    R_ASSERT(bytesRead != -1);
```
Comment (FreeZoneMods) [inline src/xrCore/FS.cpp:186]: Maybe it's better to use >= 0 condition here?

File: src/xrCore/FS.cpp [hunk 26]
```diff
@@ -181,7 +182,8 @@ void* FileDecompress(const char* fn, const char* sign, u32* size)
 
     int H = _open(fn, O_BINARY | O_RDONLY);
     R_ASSERT2(H > 0, fn);
-    _read(H, &F, 8);
+    int bytesRead = _read(H, &F, 8);
```
Comment (FreeZoneMods) [inline src/xrCore/FS.cpp:185]: Please suggest replacing magic value 8 in all places to a named constant.

File: src/xrCore/_std_extensions.h [hunk 27]
```diff
@@ -265,6 +275,7 @@ inline void MemFill32(void* dst, u32 value, size_t dstSize)
         *ptr++ = value;
 }
 
+
```
Comment (FreeZoneMods) [inline src/xrCore/_std_extensions.h:278]: remove extra line please

File: src/xrCore/xrCore.cpp [hunk 28]
```diff
@@ -231,7 +231,8 @@ void xrCore::Initialize(pcstr _ApplicationName, pcstr commandLine, LogCallback c
 #if defined(WINDOWS)
         GetCurrentDirectory(sizeof(WorkingPath), WorkingPath);
 #else
-        getcwd(WorkingPath, sizeof(WorkingPath));
+        char *prc = getcwd(WorkingPath, sizeof(WorkingPath));
+        R_ASSERT(prc != NULL);
```
Comment (FreeZoneMods) [inline src/xrCore/xrCore.cpp:235]: Use nullptr please

File: src/xrGame/map_location_defs.h [hunk 29]
```diff
@@ -1,15 +1,17 @@
 #pragma once
 #include "alife_abstract_registry.h"
 
-class CMapLocation;
+//class CMapLocation;
```
Comment (FreeZoneMods) [inline src/xrGame/map_location_defs.h]: Maybe delete this garbage instead of commenting it?

File: src/xrGame/string_table.cpp [hunk 30]
```diff
@@ -3,7 +3,10 @@
 #include "xrUICore/XML/xrUIXmlParser.h"
 #include "xr_level_controller.h"
 
-CStringTable& StringTable() { return *((CStringTable*)gStringTable); }
+CStringTable& StringTable()
+{
+    return *((CStringTable*)gStringTable);
```
Comment (FreeZoneMods) [inline src/xrGame/string_table.cpp]: Ugly cast...

File: src/xr_3da/entry_point.cpp [hunk 31]
```diff
@@ -76,19 +76,20 @@ int APIENTRY WinMain(HINSTANCE inst, HINSTANCE prevInst, char* commandLine, int
...
         {
             size_t sum = 0;
             for(i = 1; i < argc; ++i)
-                sum += strlen(argv[i]) + 1;
+                sum += xr_strlen(argv[i]) + xr_strlen(" ") + sizeof('\0');
 
-            commandLine = (char*)malloc(sum);
-            memset(commandLine, 0, sum);
+            commandLine = (LPSTR)malloc(sum);
+            xr_memset((void*)commandLine, 0, sum);
```
Comment (FreeZoneMods) [inline src/xr_3da/entry_point.cpp]: Get rid of C-style cast

File: src/xr_3da/entry_point.cpp [hunk 32]
```diff
@@ -76,19 +76,20 @@ int APIENTRY WinMain(HINSTANCE inst, HINSTANCE prevInst, char* commandLine, int
 int main(int argc, char *argv[])
 {
     int result = EXIT_FAILURE;
-
+    static std::locale locale("");
+    LPSTR commandLine = nullptr;
```
Comment (FreeZoneMods) [inline src/xr_3da/entry_point.cpp]: Is there a way to remove unsafe raw pointers at all? It would 100% resolve possible bugs and leaks. Maybe it's better to use smartpointers?

File: src/xr_3da/entry_point.cpp (hunk 18 above)
Comment (FreeZoneMods) [reply src/xr_3da/entry_point.cpp]: I also don't understand the aim of this line. Remove it or at least write a comment.

File: Externals/ode/ode/src/timer.cpp (hunk 20 above)
Comment (xrombik) [reply Externals/ode/ode/src/timer.cpp]: Using standard algo goes to use large code overhead in this case.

File: src/xrCore/Compression/SubAlloc.hpp (hunk 24 above)
Comment (nitrocaster) [reply src/xrCore/Compression/SubAlloc.hpp:66]: > Maybe use std::bitset instead of this bitmagic?

What would be the benefit?

File: src/xrGame/string_table.cpp (hunk 30 above)
Comment (nitrocaster) [reply src/xrGame/string_table.cpp]: It's not only the cast. This is one-line function and it has to be kept one-line.

File: src/xr_3da/entry_point.cpp (hunk 32 above)
Comment (nitrocaster) [reply src/xr_3da/entry_point.cpp]: Impossible. The actual problem is rather that there's no consistency in using raw pointers and this has to be addressed first.

File: Externals/ode/ode/src/timer.cpp (hunk 20 above)
Comment (nitrocaster) [reply Externals/ode/ode/src/timer.cpp]: Are you really replacing post-increments with pre-increments? This was consistent across the whole code base. What is the point?

File: src/xrCore/Compression/SubAlloc.hpp (hunk 22 above)
Comment (xrombik) [reply src/xrCore/Compression/SubAlloc.hpp:290]: This whole module is standing on C-style casts.

File: src/xrCore/Compression/SubAlloc.hpp (hunk 23 above)
Comment (xrombik) [reply src/xrCore/Compression/SubAlloc.hpp]: Consider removing these ugly C-style casts, please

**Comment by nitrocaster** [comment]
@xrombik, could you please provide a brief comment for each crash/bug that is fixed?
Ultimately, the message like 'fixes for some crashes' can turn unplayable alpha build into a well tested release build.

File: src/xrCore/Compression/SubAlloc.hpp (hunk 23 above)
Comment (xrombik) [reply src/xrCore/Compression/SubAlloc.hpp:64]: This whole module is standing on C-style casts.

File: src/xrCore/Compression/SubAlloc.hpp (hunk 24 above)
Comment (xrombik) [reply src/xrCore/Compression/SubAlloc.hpp:66]: This whole module is standing on C-style.

File: src/xrGame/string_table.cpp (hunk 30 above)
Comment (xrombik) [reply src/xrGame/string_table.cpp]: Even I do not know how this expression can be replaced.

File: src/xr_3da/entry_point.cpp (hunk 32 above)
Comment (xrombik) [reply src/xr_3da/entry_point.cpp]: If this file "entry_point.cpp" rewrite on C++17?

File: Externals/ode/ode/src/timer.cpp (hunk 20 above)
Comment (xrombik) [reply Externals/ode/ode/src/timer.cpp]: No point, just my typo (i have experience in platforms where this helps to prevent some processor instruction overheads).

File: src/xr_3da/entry_point.cpp (hunk 18 above)
Comment (xrombik) [reply src/xr_3da/entry_point.cpp]: It sets default locale for current system, some how i remember this was a bug in runtime, when default locale did not sets when program starts. But this is not my addition, I believe that the one who added this line faced this problem.
https://en.cppreference.com/w/cpp/locale/locale

File: src/xrGame/string_table.cpp (hunk 30 above)
Comment (Xottab-DUTY) [reply src/xrGame/string_table.cpp]:
```cpp
*(static_cast<CStringTable*>(gStringTable))
```

Comment (xrombik) [reply src/xrGame/string_table.cpp]: Есть ещё 16 подобных выражений, но с другими типами, стоит ли их всех переделывать?

File: src/xr_3da/entry_point.cpp (hunk 18 above)
Comment (FreeZoneMods) [reply src/xr_3da/entry_point.cpp]: Looks like a most of changes of this file are NOT the fixes for bugs but the improvements of the style. Maybe split a real bugfixes and style improvements to different commits (or even PRs)?

File: Externals/ode/ode/src/timer.cpp (hunk 20 above)
Comment (FreeZoneMods) [reply Externals/ode/ode/src/timer.cpp]: Looks like a most of changes of this file are NOT the fixes for bugs but the improvements of the style. Maybe split a real bugfixes and style improvements to different commits (or even PRs)?

File: src/xrCore/Compression/SubAlloc.hpp (hunk 22 above)
Comment (FreeZoneMods) [reply src/xrCore/Compression/SubAlloc.hpp:290]: And it's not a good style. Ok, let's replace it in later commits.

File: src/xrCore/Compression/SubAlloc.hpp (hunk 24 above)
Comment (FreeZoneMods) [reply src/xrCore/Compression/SubAlloc.hpp:66]: The benefits are:
1) It's a modern C++ way
2) It's more readable
3) It's harder to make a mistake than while performing a bitmagic
However, if the replacing requires a lot of changes - let's perform it later and won't overload the PR.

File: src/xrGame/string_table.cpp (hunk 30 above)
Comment (FreeZoneMods) [reply src/xrGame/string_table.cpp]: Однозначно стоит. Если движок у нас разрабатывается на С++, то, наверно, стоит использовать предоставляемые им преимущества? Конкретно static_cast делает код читабельнее, избавляет от неоднозначности, присущей приведениям в стиле С, а также потенциально может немного ускорять компиляцию (хотя не уверен, что какой-то выигрыш будет ощутим, но все же).
Однако тут, как я смотрю, изменения снова связаны скорее не с фиксом багов, а с изменением стиля. Давай оставим в этом PR только изменения, которые реально фиксят баги? А стили, касты и прочее уже вынесем в другой PR, дабы не затруднять изучение истории.

File: src/xr_3da/entry_point.cpp (hunk 32 above)
Comment (FreeZoneMods) [reply src/xr_3da/entry_point.cpp]: I mean remove 'unsafe' pointers in this particular fragment.  I think if you are modifying the code you should modify it qualitatively. We shouldn't make a modified code more complicated than it was before the modifications. Now we could see raw pointers, memsets, direct memory alloc and free, cast-magic... It's a very 'buggy' style, isn't it? It could be acceptably for C-styled code, but CPP-styled code should avoid such dangerous operations if it's possible. This fragment doesn't look like a performance-critical, so suggest rewriting the code in a more clear way. Maybe use std::string?

Comment (xrombik) [reply src/xr_3da/entry_point.cpp]: Какую проблему мы этим решим? Использование новых конструкци С++17 как мы видим, никак не помогает уйти от утечек памяти или нарушения сегментации.

**Comment by collectioner** [comment]
Closed due to outdated changes.
